src/index.js

```diff
@@ -251,7 +257,7 @@ export default function ({types: t}) {
                     // `<FormattedMessage id={dynamicId} />`, because it will be
                     // skipped here and extracted elsewhere. The descriptor will
                     // be extracted only if a `defaultMessage` prop exists.
-                    if (descriptor.defaultMessage) {
+                    if (descriptor.defaultMessage || opts.optionalDefaultMessages) {
```

Comment:
This breaks the plugin when spreads are used. e.g., `defineMessages` test case fails:
```
import React, {Component} from 'react';
import {defineMessages, FormattedMessage} from 'react-intl';

const msgs = defineMessages({
    header: {
        id: 'foo.bar.baz',
        defaultMessage: 'Hello World!',
        description: 'The default message',
    },
    content: {
        id: 'foo.bar.biff',
        defaultMessage: 'Hello Nurse!',
        description: 'Another message',
    },
});

export default class Foo extends Component {
    render() {
        return (
            <div>
                <h1><FormattedMessage {...msgs.header}/></h1>
                <p><FormattedMessage {...msgs.content}/></p>
            </div>
        );
    }
}
```

A proper fix could be checking for `id` presence instead. See: https://github.com/Doctoright/babel-plugin-react-intl/commit/2e13607b63b3ae4aa911a246992ef2c8a4f299af